packages/fxa-shared/subscriptions/stripe.ts

```diff
@@ -255,7 +255,12 @@ export const getSubscriptionUpdateEligibility: (
   if (
     currentPlan.plan_id === newPlan.plan_id ||
     !currentPlanConfig.productSet ||
-    currentPlanConfig.productSet !== newPlanConfig.productSet ||
+    !newPlanConfig.productSet ||
+    !currentPlanConfig.productSet.length ||
+    !newPlanConfig.productSet.length ||
+    !newPlanConfig.productSet.filter((elem) =>
+      currentPlanConfig.productSet!.includes(elem)
+    ).length ||
```

Comment:
The intent of the code here is pretty hard to understand at a glance, can this be pulled into a separate more obviously named function with a doc string that makes it clear what this check is intended to do?

Reply:
So I was going down the road of moving it into its own function but it ended up being a little messier with having to around multiple parameters and type checking (the methods that extract the configs dont return named types) and I figured i was going to have to document in that method any way what each conditional check was doing and why so i added comments for each in the original method.

let me know if that suffices. I can pull it out into its own function if we want to clean this one up but the function it gets pulled out to will be equally messy and this is so inherent to this function that the resulting one probably wouldn't be used anywhere else. i'm open to whatever though.